hazelcast/src/test/java/com/hazelcast/client/AuthenticationInformationLeakTest.java

```diff
@@ -0,0 +1,178 @@
+import com.hazelcast.internal.util.HashUtil;
+import com.hazelcast.test.HazelcastSerialClassRunner;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.UUID;
+
+import static com.hazelcast.internal.nio.Protocols.CLIENT_BINARY;
+import static com.hazelcast.test.Accessors.getClientEngineImpl;
+import static com.hazelcast.test.HazelcastTestSupport.assertContains;
+import static com.hazelcast.test.HazelcastTestSupport.assertInstanceOf;
+import static com.hazelcast.test.HazelcastTestSupport.assertNotContains;
+import static com.hazelcast.test.HazelcastTestSupport.randomString;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+
+@RunWith(HazelcastSerialClassRunner.class)
+@Category({ QuickTest.class })
+public class AuthenticationInformationLeakTest {
+
+    private static byte serializationVersion;
+    private HazelcastInstance instance;
+    private String clusterName;
+
+    @BeforeClass
+    public static void setupClass() {
+        InternalSerializationService ss = new DefaultSerializationServiceBuilder().build();
+        serializationVersion = ss.getVersion();
+    }
+
+    @Before
+    public void setup() {
+        clusterName = randomString();
+        Config config = new Config();
+        config.setClusterName(clusterName);
+        instance = Hazelcast.newHazelcastInstance(config);
+    }
+
+    @After
+    public void cleanUp() {
+        Hazelcast.shutdownAll();
+    }
+
+    @Test
+    public void testAuthenticationExceptionDoesNotLeakInfo() {
+        SerializationService ss = new DefaultSerializationServiceBuilder().build();
+        InetSocketAddress endpoint = new InetSocketAddress("127.0.0.1", 5701);
+        try (Socket socket = new Socket()) {
+            socket.setReuseAddress(true);
+            socket.connect(endpoint);
+            ClientMessage res;
+            try (OutputStream os = socket.getOutputStream(); InputStream is = socket.getInputStream()) {
+                os.write(CLIENT_BINARY.getBytes(StandardCharsets.UTF_8));
+                res = getKeyValue(ss, os, is);
+            }
+            assertEquals(res.getMessageType(), ErrorsCodec.EXCEPTION_MESSAGE_TYPE);
+            ClientExceptionFactory factory = new ClientExceptionFactory(false, Thread.currentThread().getContextClassLoader());
+            throw factory.createException(res);
+        } catch (Throwable err) {
+            String message = err.getMessage();
+            assertInstanceOf(AuthenticationException.class, err.getCause());
```

Comment:
Because the error is thrown as like [this](https://github.com/hazelcast/hazelcast/blob/e1e64194ff355f730b570bdd79c34d121394eed2/hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/AbstractMessageTask.java#L179)